pkg/clustermanager/cluster_manager.go

```diff
@@ -122,6 +122,8 @@ type ClusterClient interface {
 	SaveLog(ctx context.Context, cluster *types.Cluster, deployment *types.Deployment, fileName string, writer filewriter.FileWriter) error
 	GetMachines(ctx context.Context, cluster *types.Cluster, clusterName string) ([]types.Machine, error)
 	GetClusters(ctx context.Context, cluster *types.Cluster) ([]types.CAPICluster, error)
+	PauseCAPICluster(ctx context.Context, cluster, kubeconfig string) error
+	ResumeCAPICluster(ctx context.Context, cluster, kubeconfig string) error
```

Comment:
I think we need to implement these two through the retrier client, so they get retried in case of transient errors.